### PR TITLE
Platform name api

### LINKED ISSUE #365: API to get supported platforms names
Current only way to fetch all supported target names are like this:

```
mbeds = mbed_lstools.create()
db_items = list(mbeds.plat_db.items())
supported_targets = sorted([i[1]["platform_name"] for i in db_items])
```

Like you see it's not very simple and clear way. I would like to have dedicated public API to achieve same, eg `mbeds.get_supported_targets()` which could do these tricks for me.

### PR DESCRIPTION
**Pull request by bridadan**
This one fixes #365.

FYI @jupe

### CODE REVIEW

File: mbed_lstools/lstools_base.py [hunk 1]
```diff
@@ -403,6 +403,18 @@ def get_dummy_platform(self, platform_name):
         self.dummy_counter[platform_name] += 1
         return platform
 
+    def get_supported_platforms(self, device_type=None):
+        """! Return a dictionary of supported target ids and the corresponding platform name
+        @param device_type Filter which device entries are returned from the platform database
+        @return Dictionary of { 'target_id': 'platform_name', ... }
```
Comment (jupe) [inline mbed_lstools/lstools_base.py]: is there some needs for target_id’s? Isn’t it enough for our tools to get just platform_names? (at least mbed-flasher does not need target-ids) function names says ”platforms” so at least I would expect to get just platform_names in array 🤔

File: mbed_lstools/lstools_base.py [hunk 2]
```diff
@@ -403,6 +403,18 @@ def get_dummy_platform(self, platform_name):
         self.dummy_counter[platform_name] += 1
         return platform
 
+    def get_supported_platforms(self, device_type=None):
```
Comment (jupe) [inline mbed_lstools/lstools_base.py]: thanks for device type filtering - mbed-flasher will need it 👍

File: mbed_lstools/lstools_base.py (hunk 1 above)
Comment (bridadan) [reply mbed_lstools/lstools_base.py]: I mostly did it to not create two APIs for fetching platform names and target ids.

Comment (jupe) [reply mbed_lstools/lstools_base.py]: That’s fine for me.

**Pull request merged**